index.js

```diff
@@ -1,36 +1,47 @@
-var jsonMask = require('json-mask'),
-  compile = jsonMask.compile,
-  filter = jsonMask.filter;
+const jsonMask = require('json-mask')
+
+const badCode = code => code >= 300 || code < 200
 
 module.exports = function (opt) {
-  opt = opt || {};
+  opt = opt || {}
 
   function partialResponse(obj, fields) {
-    if (!fields) return obj;
-    return filter(obj, compile(fields));
+    if (!fields) return obj
+    return jsonMask(obj, fields)
   }
 
   function wrap(orig) {
     return function (obj) {
-      var param = this.req.query[opt.query || 'fields'];
+      const param = this.req.query[opt.query || 'fields']
       if (1 === arguments.length) {
-        orig(partialResponse(obj, param));
-      } else if (2 === arguments.length) {
-        if ('number' === typeof arguments[1]) {
-          orig(arguments[1], partialResponse(obj, param));
+        if(badCode(this.statusCode)) {
```

Comment:
Just to make sure I completely understand this, is the idea that if there's an error then we don't run it through `json-mask` because the shape of the error will be different from what the client expected?